decentraland/screens/ListLandForSaleScreen.js

```diff
@@ -8,40 +8,72 @@ import {
 import PropTypes from "prop-types";
 import LandForSaleList from "@presentational/LandForSaleList";
 import LandForSaleListItem from "@presentational/LandForSaleListItem";
-import { addressesAreEqual, showError, showInfo, getContracts } from "@helpers";
+import { showError, showInfo, getContracts, addressesAreEqual } from "@helpers";
+import { generateAssetFromId } from "@helpers/decentraland";
 import { Root } from "native-base";
-
 import DecentralandUtils from "tasit-sdk/dist/helpers/DecentralandUtils";
 
-import AssetTypes from "@constants/AssetTypes";
-const { ESTATE, PARCEL } = AssetTypes;
-
 export class ListLandForSaleScreen extends React.Component {
   componentDidMount = async () => {
+    const {
+      appendLandForSaleToList,
+      setLoadingAssetsForSaleInProgress,
+    } = this.props;
     try {
       showInfo("Loading land for sale...");
-      await this._loadAssetsForSale();
+      const assetsForSale = await this._getAllAssetsForSale();
+
+      // Non-functional style
+      // const loadingAssetsOnScreen = [];
+      // for (let promise of assetsForSale) {
+      //   let loadAssetToScreen = async () => {
+      //     const assetForSale = await promise;
+      //     appendLandForSaleToList(assetForSale);
+      //   };
+      //   loadingAssetsOnScreen.push(loadAssetToScreen());
+      // }
+
+      // Functional style
+      // const loadingAssetsOnScreen = assetsForSale.map(promise => {
+      //   let loadAssetOnScreen = async () => {
+      //     const assetForSale = await promise;
+      //     appendLandForSaleToList(assetForSale);
```

Comment:
I think I personally find this style (functional style with named function) the most readable, but others could certainly disagree for valid reasons. Do you have a strong preference?

Reply:
No strong opinion, done.

Reply:
Okay, thanks! Resolving this.